.github/workflows/run_release.yaml

```diff
@@ -9,58 +9,77 @@ on:
         description: "Release Version (e.g., 1.2.4)"
         required: true
         type: string
+      runQuickCheck:
+        description: "Run a quick check of release"
+        required: false
+        type: boolean
+        default: true
 
 name: Build and Sign Release
 
 jobs:
   build-and-deploy:
     runs-on: windows-latest
+    environment: Development
     env:
-      CODESIGN_PW: ${{ secrets.CODESIGN_PW }}
-      CODESIGN_PFX: ${{ secrets.CODESIGN_PFX }}
       RELEASE_VERSION: ${{ inputs.version }}
     permissions:
+      id-token: write
       contents: write
     steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-      with:
-        path: repo
-    - name: Sign Scripts
-      shell: pwsh
-      run: |
-        $ErrorActionPreference = "Stop"
-        $PSDefaultParameterValues['*:ErrorAction']='Stop'
-        Set-StrictMode -Version Latest
-
-        New-Item -ItemType directory -Path certificate
-        Set-Content -Path certificate\cert.txt -Value $env:CODESIGN_PFX
-        certutil -decode certificate\cert.txt certificate\cert.pfx
-
-        $cert = Get-PfxCertificate -FilePath certificate\cert.pfx -Password (ConvertTo-SecureString -String $env:CODESIGN_PW -Force -AsPlainText)
-
-        Get-ChildItem -Recurse -Path repo -Include **.ps1,**.psm1,**.psd1 | ForEach-Object {
-          $path = $_.FullName
-          Set-AuthenticodeSignature -Certificate $cert -FilePath $path -TimestampServer "http://timestamp.digicert.com/" -IncludeChain NotRoot -HashAlgorithm SHA256
-          # Delay for 4 seconds to avoid exceeding rate limits (1000 / 5 minutes, 100 / 5 seconds)
-          Start-Sleep -Seconds 4
-        }
-
-        Remove-Item -Recurse -Force certificate
-        Remove-Item -Recurse -Force repo -Include .git*
-
-        Move-Item  -Path repo -Destination "ScubaGear-${env:RELEASE_VERSION}" -Force
-        Compress-Archive -Path "ScubaGear-${env:RELEASE_VERSION}" -DestinationPath "ScubaGear-${env:RELEASE_VERSION}.zip"
-
-        Get-ChildItem -Path . | Write-Output
-    - name: release
-      uses: softprops/action-gh-release@v1
-      id: create_release
-      with:
-        draft: true
-        prerelease: false
-        name: ${{ inputs.releaseName }}
-        tag_name: v${{ inputs.version }}
-        files: ScubaGear-${{ inputs.version }}.zip
-        generate_release_notes: true
-        fail_on_unmatched_files: true
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: repo
+      - name: OIDC Login to Azure Public Cloud with AzPowershell (enableAzPSSession true)
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+          enable-AzPSSession: true       
+      - name: Sign Module
+        uses: azure/powershell@v1
+        with:
```

Comment:
(optional for now, but please file an issue if you choose to defer) Should we run the smoke test and unit tests here as part of the build workflow?   We can discuss as a team the level of testing we think is appropriate but it seems to me at least running those minimal tests automatically would be good given they are in place and relatively stable at this point.

Reply:
I do agree testing is appropriate at this point.  I did add a "quick check" later in the workflow.  After this work matures I see us updating the automated Product Tests to use a signed, deployed scuba module for testing.  So I recommend we defer.  Added #796